subcommand/health-sync/checks.go

```diff
@@ -0,0 +1,255 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package healthsync
+
+import (
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/hashicorp/consul-ecs/awsutil"
+	"github.com/hashicorp/consul-ecs/config"
+	"github.com/hashicorp/consul/api"
+	"github.com/hashicorp/go-multierror"
+)
+
+// fetchHealthChecks fetches the Consul health checks for both the service
+// and proxy registrations
+func (c *Command) fetchHealthChecks(consulClient *api.Client, taskMeta awsutil.ECSTaskMeta) (map[string]*api.HealthCheck, error) {
```

Comment:
This is a newly added function. Others in this PR are a copy paste of code removed in https://github.com/hashicorp/consul-ecs/pull/207